plugin.video.srgssr_ch_replay/resources/lib/plugin.py

```diff
@@ -0,0 +1,537 @@
+    ):
+        """Lists the latest episodes of a TV Show
+        :param tv_show_id: The id of the TV Show
+        :param current_page: Index of the current episodes page
+        :param number_of_episodes: Total number of episodes of the show
+        :param next_page_id: ID of the next page of episodes
+        """
+        xbmcplugin.setContent(self.HANDLE, "episodes")
+
+        number_of_episodes_per_page = int(self.settings.number_of_episodes_per_page)
+        res = self.video_client.get_latest_episodes(
+            self.bu, tv_show_id, page_size=number_of_episodes_per_page, next_page_id=next_page_id
+        )
+
+        show = res.get("show")
+        episodes = res.get("episodeList")
+
+        if show and episodes:
+            for episode in episodes:
+                media = episode.get("mediaList")[0]
+                self._add_video_to_directory(show, episode, media)
+
+            next_page_url = res.get("next")
+            if next_page_url:
+                self._add_next_page_to_directory(
+                    current_page,
+                    next_page_url,
+                    number_of_episodes,
+                    "list_episodes_by_show",
+                    {"tv_show_id": quote_plus(show.get("id"))},
+                )
+        xbmcplugin.endOfDirectory(self.HANDLE)
+
+    def trending(self, current_page: int, next_page_id=""):
+        """Menu listing the Trending videos"""
+        number_of_episodes_per_page = int(self.settings.number_of_episodes_per_page)
+        res = self.video_client.get_trendings(self.bu, number_of_episodes_per_page, next_page_id)
+
+        for media in res.get("mediaList"):
+            show = media.get("show")
+            episode = media.get("episode")
+            self._add_video_to_directory(show, episode, media)
+
+        next_page_url = res.get("next")
+        if next_page_url:
+            self._add_next_page_to_directory(
+                current_page,
+                next_page_url,
+                0,
+                "trending",
+            )
+        xbmcplugin.endOfDirectory(self.HANDLE)
+
+    def search_menu(self, search_type: str = ""):
+        """Search menu.
+        :param search_type: Either "tv_shows", "videos", or empty.
+                            If empty, listing the type available type of search. Else, showing a keyboard to search.
+        """
+        if not search_type:
+            self._add_item_to_directory(
+                self.tr(30031),
+                self.router.url(mode="search", **{"type": "tv_shows"}),
+                is_folder=True,
+            )
+            self._add_item_to_directory(
+                self.tr(30032),
+                self.router.url(mode="search", **{"type": "videos"}),
+                is_folder=True,
+            )
+        else:
+            if search_type == "tv_shows":
+                search_string = xbmcgui.Dialog().input(self.tr(30031))
```

Comment:
Minor issue:
When searching TV shows or broadcasts there is an add-on/Kodi error for these two scenarios:

    Pressing "cancel" => should just return
    Pressing "OK" without any input -> should just return

Reply:
see possible solution: https://github.com/ManBehindMooN/kodi_plugin_video_srgssr_ch_replay/pull/15/commits/ff45792c9b626c0079b88a82f63261997dbda0e6

Reply:
looks good to me